### ISSUE TITLE
Parenting in removeXyz, takeXyz functions? Do we need a takeReset?

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#622]
If `addXyz` functions set `parent(),` should all `removeXyz`, `removeAllXyz`, and `takeXyz` functions unset them too?
Also, should we have a `takeReset` function family?

```
void Model::removeAllUnits()
{
    mPimpl->mUnits.clear();
}

bool Component::removeReset(size_t index)
{
    if (index < mPimpl->mResets.size()) {
        mPimpl->mResets.erase(mPimpl->mResets.begin() + int64_t(index));
        return true;
    }
    return false;
}

bool Component::removeReset(const ResetPtr &reset)
{
    auto result = mPimpl->findReset(reset);
    if (result != mPimpl->mResets.end()) {
        mPimpl->mResets.erase(result);
        return true;
    }
    return false;
}

void Component::removeAllResets()
{
    mPimpl->mResets.clear();
}

```
Thoughts @agarny @nickerso @hsorby ?

**Comment by agarny** [comment]
Definitely, not least since `Model::removeUnits()` and `Model::takeUnits()` do, for instance, unset the parent. So, yes, it looks like some oversight which has resulted in inconsistencies.

**Comment by kerimoyle** [comment]
And do we need `takeReset() `?

**Comment by agarny** [comment]
Yes, for consistency.

**Comment by kerimoyle** [comment]
This is fixed in #621

**Comment by kerimoyle** [comment]
Closed by #621
